src/common/get-file-info.js

```diff
@@ -0,0 +1,40 @@
+"use strict";
+
+const createIgnorer = require("../common/create-ignorer");
+const options = require("../main/options");
+
+/**
+ * @param {string} filePath
+ * @param {{ ignorePath?: string, withNodeModules?: boolean, plugins: object }} opts
+ *
+ * Please note that prettier.getFileInfo() expects opts.plugins to be an array of paths,
+ * not an object. A transformation from this array to an object is automatically done
+ * internally by the method wrapper. See withPlugins() in index.js.
+ */
+function _getFileInfo(filePath, opts) {
+  let ignored = false;
+  const ignorer = createIgnorer(opts.ignorePath, opts.withNodeModules);
+  ignored = ignorer.ignores(filePath);
+
+  const inferredParser = options.inferParser(filePath, opts.plugins) || null;
+
+  return {
+    ignored,
+    inferredParser
+  };
+}
+
+// the method has been implemented as asynchronous to avoid possible breaking changes in future
+function getFileInfo(filePath, opts) {
```

Comment:
This async method looks just like

```js
(...args) => Promise.resolve(_getFileInfo(...args))
```

We need to use `createIgnorer`/`createIgnorer.sync` if we'd like to make it really async.

Reply:
It's just a quick demo, not meant to be used in node 4. 😂

> does this have to be done now?

Yeah, it can be another PR, just a reminder.

Reply:
Changing the function to the following does not work:

```js
function getFileInfo(filePath, opts) {
  return Promise.resolve(_getFileInfo(filePath, opts));
}
```

It throws rather than rejects, because `_getFileInfo()` is sync.

Reply:
Opened #4452 to track this issue.